CommandLine/libAssets/assets.cpp

```diff
@@ -0,0 +1,433 @@
+/** Copyright (C) 2018 Robert B. Colton

```

Comment:
1. You have to include the original license from the code you moved to this file.
2. There's not a compelling reason to duplicate that code, here.
3. You removed all the nice logging mechanisms that code had. The mechanism was relatively well-contained; you could replace it with a no-op logger if you genuinely had nowhere to send that information, but odds are, you can have it stream to your RPC when writing to stdout or LGM isn't good enough.

The `lang_CPP` class was meant to be an implementation of a language export for C++. The other I intended was EGMjs (other repository). If the class depended on global state (such as for logging), that's relatively easy to correct. The answer here is not to duplicate that code and make it this gigantic monolithic file. Please don't do this.

Reply:
1. It is the original license, I was just missing the original author(s) until dbd36a4c7c1e683ee5a691c920c2977a3b938c97
2. We need a version that accepts a proto, not EnigmaStruct, but we're not yet ready to break LGM until the next pull request.
3. I didn't remove them, I commented them out for a second. Regardless, that's a fair point you've made. So you're saying, even when RGM calls the function directly (since it statically links `libAssets`) it should send the log messages back to the server and then to RGM through GRPC? That's going to require some additional restructuring to the logging, because the `BuildGame` RPC is already in the "finished" state when RGM writes the assets, hence there's no more opportunity to read the log messages from CompileReply. That's why I suggested offering a way for RGM to read `libAssets` log messages directly.

>The lang_CPP class was meant to be an implementation of a language export for C++.

Right, I understood this much.

>The answer here is not to duplicate that code and make it this gigantic monolithic file.

Are you talking about just the logging or the logging and asset writing? You see, you should understand by now the idea here is to let RGM write the assets to the game. If that code is embedded into `lang_cpp` then `lang_cpp` and the rest of `CompilerSource` also has to be linked into RGM for it to be able to write the assets.

What else do you propose? We've already discussed your idea of distributing an emake binary with RGM but we preferred the single-executable approach. I need additional feedback from you before I can continue.